packages/rabbitmq/src/utils.ts

```diff
@@ -0,0 +1,128 @@
+import { CSL, Cardano, cslToCore } from '@cardano-sdk/core';
+import { OutsideOfValidityInterval } from '@cardano-ogmios/schema';
+import { fromSerializableObject, toSerializableObject } from '@cardano-sdk/util';
+
+export interface DeserializedError {
+  innerError?: {
+    name: string;
+    message: string;
+  };
+  name: string;
+  message: string;
+}
+
+export interface SerializedError {
```

Comment:
Is this type useful anywhere? If not, using `unknown` in place of it would work.

Reply:
Its usefulness is to avoid using lots of `as` operators while accessing the attributes of a serializable error.
Looking at it once again, since the error is not yet serialized, but in the special form which can be serialized it's better to rename the type in `SerializableError`...
Once applied all the other changes requested in this review, I'll check how many time I'm going to use to decide if to keep it or remove it

Reply:
Do you need to access attributes of serialized error anywhere? Why not just deserialize into a proper error type before doing that?

Reply:
Probably once applied all the changes, only inside the `getErrorPrototype` function

Reply:
Ah yes, I need to access attributes in order to check if the error is retriable or not

Reply:
Or maybe not... I could perform the check against the original error...

Reply:
Right, it's probably best to serialize it just before sending